partner_compassion/models/partner_compassion.py

```diff
@@ -23,6 +23,32 @@ class ResPartner(models.Model):
     """
     _inherit = 'res.partner'
 
+    @api.multi
+    @api.depends('category_id')
+    def _compute_has_sponsorships(self):
+        """
+        A partner is sponsor if he is correspondent of at least one
+        sponsorship.
+        """
+        for partner in self:
+            if partner.is_church:
+                partner.has_sponsorships = self.env[
+                    'recurring.contract'].search_count([
+                        '|',
+                        ('partner_id', 'in', partner.member_ids.ids),
+                        ('correspondant_id', 'in', partner.member_ids.ids),
+                        ('type', 'like', 'S')
```

Comment:
You must as well include the direct sponsorships of the church, by adding the condition (with OR operator) `('partner_id', '=', partner.id)` and same for correspondant_id